frontend/src/pages/Assignments/StudentSubmission/index.tsx

```diff
@@ -0,0 +1,124 @@
+  const assignmentID = location.state.assignmentId;
+  const [firstCommit, setFirstCommit] = useState<string>("");
+  const [totalCommits, setTotalCommits] = useState<string>();
+
+  console.log(location.state);
+
+  useEffect(() => {
+    if (location.state && location.state.submission) {
+      setSubmission(location.state.submission); // Use submission from state
+    } else if (id && assignmentID && selectedClassroom) {
+      // If state fails, use submission data as a fallback
+      (async () => {
+        try {
+          const fetchedSubmission = await getStudentWorkById(
+            selectedClassroom.id,
+            assignmentID, // Use assignmentID from state
+            +id // Student submission ID
+          );
+          if (fetchedSubmission) {
+            setSubmission(fetchedSubmission);
+          }
+        } catch (error) {
+          console.error("Failed to fetch submission:", error);
+        }
+      })();
+    }
+  }, [location.state, id, selectedClassroom]);
+
+  useEffect(() => {
+    if (submission !== null && submission !== undefined && selectedClassroom !== null && selectedClassroom !== undefined) {
+      (async () => {
+        try {
+          const commitDate = await getFirstCommit(
+            selectedClassroom.id,
+            assignmentID,
+            submission.student_work_id
+          );
+          if (commitDate !== null && commitDate !== undefined) {
+            setFirstCommit(formatDate(commitDate));
+          } else {
+            setFirstCommit("N/A");
+          }
+        } catch (_) {
+          // do nothing
+        }
+      })();
+    }
+  }, [selectedClassroom, submission]);
+
+  useEffect(() => {
+    if (submission !== null && submission !== undefined && selectedClassroom !== null && selectedClassroom !== undefined) {
+      (async () => {
+        try {
+          const total = await getTotalCommits(
+            selectedClassroom.id,
+            assignmentID,
+            submission.student_work_id
+          );
+
+          console.log(total);
+          if (totalCommits !== null && totalCommits !== undefined) {
+            setTotalCommits(total.toString());
+          } else {
+            setTotalCommits("N/A");
+          }
+
+        } catch (_) {
+          // do nothing
+        }
+      })();
+    }
+  }, [selectedClassroom, submission]);
+
+  return (
+    <div className="StudentWork">
+      <SubPageHeader
+        pageTitle={submission?.contributors.join(", ")}
+        pageSubTitle={submission?.assignment_name}
+        chevronLink={`/app/assignments/${assignmentID}`}
+      >
+      </SubPageHeader>
+
+      <div className="StudentSubmission__externalButtons">
+        <Button href="#" variant="secondary" newTab>
+          <FaGithub className="icon" /> View Student Repository
+        </Button>
+        <Button href="#" variant="secondary" newTab>
+          <MdEditDocument className="icon" />  Grade Submission
```

Comment:
Button breaks page because of state issue

Reply:
done